pycbc/io/live.py

```diff
@@ -368,13 +359,10 @@ def upload(self, fname, gracedb_server=None, testing=True,
                 curr_snrs.save(snr_series_fname, group='%s/snr' % ifo)
                 pl.plot(curr_snrs.sample_times - ref_time, abs(curr_snrs),
                         c=ifo_color(ifo), label=ifo)
-                if ifo in self.ifos:
-                    base = 'foreground/{}/'.format(ifo)
-                    snr = self.coinc_results[base + 'snr']
-                    mt = (self.coinc_results[base + 'end_time']
-                          + self.time_offset)
-                    pl.plot([mt - ref_time], [snr], c=ifo_color(ifo),
-                            marker='x')
+                base = 'foreground/{}/'.format(ifo)
+                snr = self.coinc_results[base + 'snr']
+                mt = (self.coinc_results[base + 'end_time'] + self.time_offset)
+                pl.plot([mt - ref_time], [snr], c=ifo_color(ifo), marker='x')
```

Comment:
I think here you are again assuming that each `snr_ifo` will have a corresponding entry in `coinc_results`. This is the case right now for PyCBC Live, but may not be in the future (e.g. KAGRA used only for BAYESTAR but not for improving the FAR) or for other applications.

I would not make this change.